### PR TITLE
feat: Use Imager::zxing to read barcodes from uploaded images

### LINKED ISSUE #8925: Update barcode recognition to understand more barcode types
### What
- Barcode parsing should understand more barcode formats, like DataMatrix and DataBar, and code normalization should be able to extract the GTIN from a GS1-128, GS1 DataMatrix, a GS1 QR Code, a GS1 Data Link, or a GS1 AI data string. 
- This will enable users to use those barcodes, which will become the norm with the GS1 "Sunrise 2027" initiative, in search.

### Part of
- #8333

### PR DESCRIPTION
**Pull request by hangy**
### What

Use Perl module `Imager::zxing` to read barcodes from images to enable support for more barcode types that are not supported by `Barcode::ZBar`, ie. DataMatrix. By passing the values to `GS1::SyntaxEngine::FFI`, this allows GS1 Data URI, GS1 DataMatrix, and GS1 QR Code barcodes to be used & parsed when uploading images to Product Opener.

### Related issue(s) and discussion

- Fixes #8925
- Includes #9049

### CODE REVIEW

**Comment by teolemon** [comment]
@hangy @alexgarel WDYT ?

**Comment by hangy** [comment]
@teolemon The change depends on the moving to the new(?) server (new Debian version)

File: tests/unit/images.t [hunk 1]
```diff
@@ -40,4 +40,27 @@ foreach my $test_ref (@tests) {
...
+	["01_upc_a.jpg", "0725272730706"],
+	["02_upc_e.jpg", "01234565"],
+	["03_itf_13.jpg", "0012345123456"],
+	["04_gs1_128.jpg", "4044782317112"],
+	["05_gs1_databar_omni.jpg", "5010029000214"],
+	["07_gs1_databar_stacked.jpg", "5010029000214"],
+	["08_gs1_databar_stacked_omni.jpg", "5010029000214"],
+	["37_gs1_datamatrix.jpg", "0725272730706"],
+	["43_gs1_qrcode_digital_link.jpg", "9506000134369"],
+	["52_gs1_datamatrix_digital_link.jpg", "5010029000214"]
```
Comment (alexgarel) [inline tests/unit/images.t:56]:
```suggestion
	# extracted from https://id.gs1.org/01/05010029000214/10/AbC/21/666?17=301010
	["52_gs1_datamatrix_digital_link.jpg", "5010029000214"]
```

File: tests/unit/images.t [hunk 2]
```diff
@@ -40,4 +40,27 @@ foreach my $test_ref (@tests) {
...
+
+	["01_upc_a.jpg", "0725272730706"],
+	["02_upc_e.jpg", "01234565"],
+	["03_itf_13.jpg", "0012345123456"],
+	["04_gs1_128.jpg", "4044782317112"],
+	["05_gs1_databar_omni.jpg", "5010029000214"],
+	["07_gs1_databar_stacked.jpg", "5010029000214"],
+	["08_gs1_databar_stacked_omni.jpg", "5010029000214"],
+	["37_gs1_datamatrix.jpg", "0725272730706"],
+	["43_gs1_qrcode_digital_link.jpg", "9506000134369"],
```
Comment (alexgarel) [inline tests/unit/images.t:55]:
```suggestion
	# extracted from https://dalgiardino.com/01/09506000134369/10/ABC/21/123456?17=230829
	["43_gs1_qrcode_digital_link.jpg", "9506000134369"],
```

**Review by alexgarel — APPROVED** [review]
Thank @hangy !
LGTM, just some minor comments or explanations needed.

**Pull request merged**